.github/workflows/publish.yml

```diff
@@ -1,21 +1,30 @@
 name: Publish crates
 
 on:
+  # We only want to perform dry runs when this workflow runs on PRs.
+  pull_request:
+    branches: [master]
   workflow_dispatch:
     inputs:
       acvm-ref:
         description: The acvm reference to checkout
         required: true
+      publish:
+        description: Publish packages to crates.io
+        type: boolean
+        required: true
 
 jobs:
   publish:
     name: Publish in order
     runs-on: ubuntu-latest
+    env:
+      DRY_RUN_FLAG: ${{ (inputs.publish && "") || "--dry-run" }}
```

Comment:
From github actions
> The workflow is not valid. .github/workflows/publish.yml (Line: 22, Col: 21): Unexpected symbol: '""'. Located at position 20 within expression: (inputs.publish && "") || "--dry-run"

Do we need the `""`?

Reply:
huh, strange. I've replaced this logic to avoid the `""`, I just had it like this to be similar to an `if-else`.